res/shaders/halo.hlsl

```diff
@@ -0,0 +1,56 @@
+cbuffer object_buffer : register(b0)
+{
+    float4x4 projection_view_model;
+    float4x4 world;
+};
+
+cbuffer camera_buffer : register(b2)
+{
+    float3 camera_pos;
+    float padding1;
+    float3 obj_origin;
+    float padding2;
+}
+
+struct vs_input
+{
+    float3 pos : POSITION;
+    float3 normal : NORMAL;
+    float2 uv : TEXCOORD;
+};
+
+struct vs_output
+{
+    float4 pos : SV_POSITION;
+    float2 uv : TEXCOORD;
+    float3 world_normal : NORMAL;
+    float3 world_pos : POSITION;
+};
+
+Texture2D obj_texture;
+SamplerState obj_sampler_state;
```

Comment:
It's better to explicitly declare what Texture/Sampler register we are using. In this case to and s0.